### PR TITLE
W2-phongle-Create calculator screen

### PR DESCRIPTION
**Pull request by at-phongle**
#### Related issues (Optional)
Task # 2
#### Description (Required)
- [x] Create Main Screen
- [x] Create Calculator Screen

### CODE REVIEW

**Comment by at-phongle** [comment]
@at-huongnguyen please review this  .
Thank you so much!

File: app/src/at-phongle/java/vn/asiantech/internship/app_calculator/CalculatorActivity.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.calculator_activity);
+        initViews();
+        mEdtNumberA.addTextChangedListener(this);
+        mEdtNumberB.addTextChangedListener(this);
+        mBtnAdd.setOnClickListener(this);
+        mBtnSub.setOnClickListener(this);
+        mBtnMul.setOnClickListener(this);
+        mBtnDiv.setOnClickListener(this);
```
Comment (at-huongnguyen) [inline app/src/at-phongle/java/vn/asiantech/internship/app_calculator/CalculatorActivity.java]: Line 30~35, should create initListener method

File: app/src/at-phongle/java/vn/asiantech/internship/app_calculator/CalculatorActivity.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        mBtnDiv = findViewById(R.id.btDiv);
+        mTvOperator = findViewById(R.id.tvOperator);
+        mTvResult = findViewById(R.id.tvResult);
+    }
+
+    @Override
+    public void onClick(View v) {
+        switch (v.getId()) {
+            case R.id.btAdd:
+                mTvOperator.setText("+");
```
Comment (at-huongnguyen) [inline app/src/at-phongle/java/vn/asiantech/internship/app_calculator/CalculatorActivity.java]: Use from Strings resource

File: app/src/at-phongle/java/vn/asiantech/internship/app_calculator/CalculatorActivity.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+                mTvResult.setText(String.valueOf(operator(mEdtNumberA, mEdtNumberB, '-')));
+                break;
+            case R.id.btMul:
+                mTvOperator.setText("*");
+                mTvResult.setText(String.valueOf(operator(mEdtNumberA, mEdtNumberB, '*')));
+                break;
+            case R.id.btDiv:
+                mTvOperator.setText("/");
+                mTvResult.setText(String.valueOf(operator(mEdtNumberA, mEdtNumberB, '/')));
+                break;
```
Comment (at-huongnguyen) [inline app/src/at-phongle/java/vn/asiantech/internship/app_calculator/CalculatorActivity.java]: '+, -, *, /' Use from Strings resource

File: app/src/at-phongle/java/vn/asiantech/internship/app_calculator/CalculatorActivity.java [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+    private void setButtonEnable() {
+        mBtnAdd.setEnabled(checkInvalidNumber(mEdtNumberA, mEdtNumberB));
+        mBtnSub.setEnabled(checkInvalidNumber(mEdtNumberA, mEdtNumberB));
+        mBtnMul.setEnabled(checkInvalidNumber(mEdtNumberA, mEdtNumberB));
+        mBtnDiv.setEnabled(checkInvalidNumber(mEdtNumberA, mEdtNumberB));
+        mBtnDiv.setEnabled(checkZeroNumber(mEdtNumberB));
+    }
+
+    private float operator(EditText edtA, EditText edtB, char operator) {
```
Comment (at-huongnguyen) [inline app/src/at-phongle/java/vn/asiantech/internship/app_calculator/CalculatorActivity.java]: Should change 2 params to float or double

File: app/src/at-phongle/java/vn/asiantech/internship/app_calculator/CalculatorActivity.java [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+            case '*':
+                return numA * numB;
+            case '/':
+                return numA / numB;
+            default:
+                return 0;
+        }
+    }
+
+    private boolean checkInvalidNumber(EditText edtA, EditText edtB) {
```
Comment (at-huongnguyen) [inline app/src/at-phongle/java/vn/asiantech/internship/app_calculator/CalculatorActivity.java]: Should use 1 param and change EditText to String

File: app/src/at-phongle/java/vn/asiantech/internship/app_calculator/CalculatorActivity.java [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+    @Override
+    public void onTextChanged(CharSequence s, int start, int before, int count) {
+        setButtonEnable();
+    }
+
+    @Override
+    public void afterTextChanged(Editable s) {
+        setButtonEnable();
+    }
```
Comment (at-huongnguyen) [inline app/src/at-phongle/java/vn/asiantech/internship/app_calculator/CalculatorActivity.java]: Should remove `setButtonEnable` call in `beforeTextChanged ` and  `afterTextChanged ` method

File: app/src/at-phongle/res/layout/activity_app.xml [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:background="@drawable/bg_main"
+    android:gravity="center"
+    android:orientation="vertical"
+    android:padding="50dp">
+
+    <Button
+        android:id="@+id/btn_login_screen"
```
Comment (at-huongnguyen) [inline app/src/at-phongle/res/layout/activity_app.xml]: btnLoginScreen

File: app/src/at-phongle/res/layout/activity_app.xml [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text="@string/btn_login_screen" />
+
+    <Button
+        android:id="@+id/btn_calculator_screen"
```
Comment (at-huongnguyen) [inline app/src/at-phongle/res/layout/activity_app.xml]: btnCalculatorScreen

File: app/src/at-phongle/res/layout/calculator_activity.xml [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+    </LinearLayout>
+
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="10dp"
+        android:orientation="horizontal">
+
+        <Button
+            android:id="@+id/btAdd"
```
Comment (at-huongnguyen) [inline app/src/at-phongle/res/layout/calculator_activity.xml]: btnAdd

File: app/src/at-phongle/res/layout/calculator_activity.xml [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+            android:layout_width="0dp"
+            android:layout_height="45dp"
+            android:layout_weight="1"
+            android:enabled="false"
+            android:text="@string/btn_add"
+            android:textSize="20dp"
+            tools:ignore="ButtonStyle" />
+
+        <Button
+            android:id="@+id/btSub"
```
Comment (at-huongnguyen) [inline app/src/at-phongle/res/layout/calculator_activity.xml]: btnSub

File: app/src/at-phongle/res/layout/calculator_activity.xml [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+            android:layout_width="0dp"
+            android:layout_height="45dp"
+            android:layout_weight="1"
+            android:enabled="false"
+            android:text="@string/btn_sub"
+            android:textSize="20dp"
+            tools:ignore="ButtonStyle" />
+
+        <Button
+            android:id="@+id/btMul"
```
Comment (at-huongnguyen) [inline app/src/at-phongle/res/layout/calculator_activity.xml]: btnMul

File: app/src/at-phongle/res/layout/calculator_activity.xml [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+            android:layout_width="0dp"
+            android:layout_height="45dp"
+            android:layout_weight="1"
+            android:enabled="false"
+            android:text="@string/btn_mul"
+            android:textSize="20dp"
+            tools:ignore="ButtonStyle" />
+
+        <Button
+            android:id="@+id/btDiv"
```
Comment (at-huongnguyen) [inline app/src/at-phongle/res/layout/calculator_activity.xml]: btnDiv

**Comment by at-phongle** [comment]
@at-huongnguyen  please review this pull request.Thanks

**Comment by at-phongle** [comment]
@at-huongnguyen  please review this . Thanks

**Comment by at-huongnguyen** [comment]
@at-phongle 
Plz resolve conflicts

**Comment by at-phongle** [comment]
@at-huongnguyen  I fixed conflix , please review this.Thanks

**Comment by at-phongle** [comment]
@at-huongnguyen  i fixed logic error. Please review this . Thanks

**Pull request merged**